test/suites/common/tx-pause/test_maintenance_mode.ts

```diff
@@ -0,0 +1,93 @@
+import "@tanssi/api-augment";
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import { DANCE } from "util/constants";
+import { ApiPromise } from "@polkadot/api";
+import { KeyringPair } from "@moonwall/util";
+import { Result } from "@polkadot/types-codec";
+import { SpRuntimeDispatchError } from "@polkadot/types/lookup";
+
+describeSuite({
+    id: "C0401",
+    title: "Pausing is compatible with maintenance mode",
+    foundationMethods: "dev",
+    testCases: ({ it, context }) => {
+        let polkadotJs: ApiPromise;
+        let alice: KeyringPair;
+        let bob: KeyringPair;
+        let chain: string;
+
+        beforeAll(async function () {
+            polkadotJs = context.polkadotJs();
+            alice = context.keyring.alice;
+            bob = context.keyring.bob;
+            chain = polkadotJs.consts.system.version.specName.toString();
+        });
+
+        it({
+            id: "E01",
+            title: "a paused tx should still fail during maintenance mode",
+            test: async function () {
+                // Pause Balances.transfer_allow_death
+                const { result } = await context.createBlock(
+                    polkadotJs.tx.sudo
+                        .sudo(polkadotJs.tx.txPause.pause(["Balances", "transfer_allow_death"]))
```

Comment:
The API is a lot nicer than what I expected! What happens if you make a typo here? Does it return an error?

Reply:
There must be some way to programatically get the correct name. I found this for the extrinsic name:

```
const tx = polkadotJs.tx.balances.transferAllowDeath(bob.address, DANCE);
console.log(tx.meta.name.toJSON());
```

No idea how to get the pallet name.

But since in polkadot js you have to enter the name manually anyway, let's leave the test with the full name, at least it documents the correct way to call this.

Reply:
That was a good lead, but I couldn't find a way to get the pallet name either. Leaving it as is, then